src/fusor/translator.py

```diff
@@ -836,3 +865,119 @@ async def from_genie(
             ce,
             rf,
         )
+
+    ######### Knowledgebase -> FUSOR CategoricalFusion object #############
+    def __process_vicc_nomenclature(self, gene_symbol: str) -> str:
+        """Extract fusion partner from VICC nomenclature
+
+        :param gene_symbol: The unprocessed gene symbol
+        :return The processed gene symbol
+        """
+        if "entrez" in gene_symbol:
+            return gene_symbol.split("(")[0]
+        start = gene_symbol.find("(")
+        stop = gene_symbol.find(")")
+        return gene_symbol[start + 1 : stop]
+
+    async def from_civic(self, civic: CIVIC) -> CategoricalFusion:
+        """Convert CIViC record to Categorical Fusion
+
+        :param civic A CIVIC object
+        :return A CategoricalFusion object, if construction is successful
+        """
+        if not isinstance(
+            civic.five_prime_end_exon_coordinates, ExonCoordinate
+        ) and not isinstance(civic.three_prime_start_exon_coordinates, ExonCoordinate):
+            msg = "Coordinate location is not available for either fusion partner"
+            _logger.warning(msg)
+            return msg
```

Comment:
Should we be raising an exception here?

Reply:
I'm going to remove this check since we want to be able to create CategoricalFusions even when the transcript data is not available (these fusions still have linked `MolecularProfiles`).